CRM/Extension/Browser.php

```diff
@@ -157,21 +157,31 @@ public function getExtensions() {
 
   /**
    * Get a description of a particular extension.
+   * This function is an optimised form of getExtensions() - it returns the
+   * extension's details as soon as it's found without caching anything.
    *
    * @param string $key
    *   Fully-qualified extension name.
    *
    * @return CRM_Extension_Info|NULL
    */
   public function getExtension($key) {
-    // TODO optimize performance -- we don't need to fetch/cache the entire repo
-    $exts = $this->getExtensions();
-    if (array_key_exists($key, $exts)) {
-      return $exts[$key];
+    if (!$this->isEnabled() || count($this->checkRequirements())) {
+      return NULL;
     }
-    else {
+
+    $remote = $this->_discoverRemote();
```

Comment:
My point was more that the original comment was

> // TODO optimize performance -- we don't need to fetch/cache the entire repo

In my mind this sounds like by only requesting details for a single extension there's no need to **fetch** all of them. Optimizing network calls is a big benefit, optimizing PHP calls in code (as this PR does) will possible offer some very small improvement, but the code is almost identical and the improvement will be small. 

Also, the original comment mentioned not **caching** the entire repo. The changed method still calls `_discoverRemote` which does the actual caching, so I don't see how this comment is fixed. 

I think it looks nicer now and it's always good to see improvements to the code, but I'm worried that the original intent to improve the performance here with the TODO comment will be lost.